config/parse.go

```diff
@@ -0,0 +1,197 @@
+// config package provides set of utilities for configuration parsing
+package config
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mholt/caddy/caddyfile"
+)
+
+type CfgTreeNode struct {
+	Name      string
+	Args      []string
+	Childrens []CfgTreeNode
+	Snippet   bool
+}
+
+type parseContext struct {
+	caddyfile.Dispenser
+	parens   int
```

Comment:
typo in `parens`

Reply:
https://www.dictionary.com/browse/parens
Is it really a typo? If my English is that bad - which word (or form) I should use then?

Reply:
sorry, I thought you meant « parents »